### DISCUSSION TITLE
How can I schedule a cron job for :adblock-update?

### DISCUSSION THREAD

**Discussion by crocket** [qutebrowser/qutebrowser#6683] [Q&A] [answered]
```
qutebrowser --nowindow :adblock-update
```

updates adblock lists, but it doesn't quit after updating adblock lists.

Thus, this is not suitable for a cron job.

Has anyone succeeded in creating a cron job for qutebrowser commands?

**Comment by The-Compiler** [comment, marked as answer]
The only thing I can think of is something like this (mostly untested):

```bash
#!/bin/bash
tmpdir=$(mktemp -d)
trap 'rm -rf "$tmpdir"' EXIT
qutebrowser --nowindow --basedir "$tmpdir" :adblock-update ":later 10000 :quit"    # might need to tweak the timeout
cp "$tmpdir/data/blocked-hosts" ~/.local/share/qutebrowser
```

Note that'll likely still require a GUI to run. Adding an `export QT_QPA_PLATFORM=offscreen` or using `xvfb-run` might help.

**Reply by crocket** [reply]
It seems that commands run concurrently. Is it possible to make commands run sequentially?

**Reply by The-Compiler** [reply]
No, see #3007
